### PR TITLE
Simple api implementation

### PR DESCRIPTION
**Pull request by fwh-dc**
Initial implementation of wolfSSL extension for oatpp.

Note:
The ```oatpp::test::wolfssl::FullTest test_port(8443, 10);``` sometimes gets stuck after successfully executing the for loop of requests. I am unsure if there is anything I can do to fix that.

### CODE REVIEW

File: src/oatpp-wolfssl/Config.cpp [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+}
+}}
+#endif
+
+namespace oatpp { namespace wolfssl {
+
+Config::Config(WOLFSSL_METHOD *protocolMethod) {
+  int res = wolfSSL_Init();
+  if (res != SSL_SUCCESS) {
+    OATPP_LOGD("[oatpp::wolfssl::client::ConnectionProvider::get()]", "Error. Call to wolfSSL_init() failed. Return value=%d", res);
```
Comment (lganzzzo) [inline src/oatpp-wolfssl/Config.cpp]: Should be `[oatpp::wolfssl::Config::Config()]`

File: src/oatpp-wolfssl/Config.cpp [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+static void wolfsslDebug(const int logLevel, const char* const logMessage) {
+  OATPP_LOGD("[wolfssl]", "[%s] %s", logLevel, logMessage)
+}
+}}
+#endif
+
+namespace oatpp { namespace wolfssl {
+
+Config::Config(WOLFSSL_METHOD *protocolMethod) {
+  int res = wolfSSL_Init();
```
Comment (lganzzzo) [inline src/oatpp-wolfssl/Config.cpp]: Is it safe to call this method multiple times?
[wolfSSL_Init](https://www.wolfssl.com/doxygen/group__TLS.html#ga789ef74e34df659a62f06da2ea709737)

File: src/oatpp-wolfssl/Connection.cpp [hunk 3]
```diff
@@ -0,0 +1,217 @@
...
+  auto sslSession = m_connection->m_sslSession;
+  auto result = 0;
+
+  while ((result = wolfSslHandshake(sslSession)) != WOLFSSL_SUCCESS)
+  {
+    auto error = wolfSSL_get_error(sslSession, result);
+    if (error != WOLFSSL_ERROR_WANT_READ || error != WOLFSSL_ERROR_WANT_WRITE)
+    {
+      char buffer[80];
+      OATPP_LOGE("[Connection::ConnectionContext::init()]", "Error. Call to wolfSSL_connect(%p): %s", sslSession, wolfSSL_ERR_error_string(error, buffer));
```
Comment (lganzzzo) [inline src/oatpp-wolfssl/Connection.cpp]: should be `[oatpp::wolfssl::Connection::ConnectionContext::init()]`

File: src/oatpp-wolfssl/client/ConnectionProvider.hpp [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+
+#include "oatpp/network/Address.hpp"
+#include "oatpp/network/ConnectionProvider.hpp"
+
+#include "wolfssl/ssl.h"
+
+namespace oatpp { namespace wolfssl { namespace client {
+
+/**
+ * MbedTLS client connection provider.
```
Comment (lganzzzo) [inline src/oatpp-wolfssl/client/ConnectionProvider.hpp]: 👀  WolfSSL

File: src/oatpp-wolfssl/server/ConnectionProvider.hpp [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+#include "oatpp-wolfssl/Connection.hpp"
+#include "oatpp-wolfssl/Config.hpp"
+
+#include "oatpp/network/Address.hpp"
+#include "oatpp/network/tcp/server/ConnectionProvider.hpp"
+
+namespace oatpp { namespace wolfssl { namespace server {
+
+/**
+ * MbedTLS server connection provider.
```
Comment (lganzzzo) [inline src/oatpp-wolfssl/server/ConnectionProvider.hpp]: 👀  WolfSSL

**Review by lganzzzo — COMMENTED** [review]
Looks great!

There are comments/questions/typos ...

**Comment by fwh-dc** [comment]
Sorry for the late response. I thought GitHub would send out emails when you replied, but that was not the case 👎

File: src/oatpp-wolfssl/Config.cpp (hunk 2 above)
Comment (fwh-dc) [reply src/oatpp-wolfssl/Config.cpp]: I was thinking that to myself when I wrote it.
I will fix it.

**Comment by fwh-dc** [comment]
Is there more to be done here or can I merge? :)

**Review by lganzzzo — APPROVED** [review]
Looks great!
Let's have it merged

**Pull request merged**